staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -620,3 +628,69 @@ func setTargetKind(obj Object, kind schema.GroupVersionKind) {
 	}
 	obj.GetObjectKind().SetGroupVersionKind(kind)
 }
+
+// this only fails on programmer error
```

Comment:
doc all specified versions must be in the same group, and the specified order overwrites any previously specified order for this group